### DISCUSSION TITLE
How to filter textArray[]

### DISCUSSION THREAD

**Discussion by nuwanvc** [supabase/supabase#11115] [Questions] [answered]
How do I filter textArray[] in the below code?

const { data, error } = await supabase.from('arraytest').select('textarray[]')

let's say my text array has the below values from the table.

["a", "b", "c" ]
[ "b", "c"]
[ "a" ] 

how could I filter ones with 'a'
and ones with 'b'

Thanks.

**Comment by GaryAustin1** [comment, marked as answer]
You use the .contains filter.   So `.contains('textarray',["a","b"])`

**Reply by nuwanvc** [reply]
@GaryAustin1 Thanks. I will check it. 
I hope it will return rows with a or b. 
Not only the rows contains. [ "a", "b" ]

**Reply by GaryAustin1** [reply]
That is a and b....
.overlaps is what you want for any of these in the array.

**Reply by nuwanvc** [reply]
@GaryAustin1 Awesome. Overlap is perfect for my need.
Thanks, man you made my day.

**Reply by GaryAustin1** [reply]
I'm not proud... click the Answered button ... have to stay 20 ahead of the Supabase employees.. LOL
